cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CapacityConfiguration.java

```diff
@@ -0,0 +1,257 @@
+ *
+ * kind: KafkaNodePool
+ * metadata:
+ *   name: controller
+ *   labels:
+ *     strimzi.io/cluster: my-cluster
+ * spec:
+ *   replicas: 3
+ *   roles:
+ *     - controller
+ *   storage:
+ *     type: jbod
+ *     volumes:
+ *       - id: 0
+ *         type: ephemeral
+ *         kraftMetadata: shared
+ * ---
+ *
+ * apiVersion: kafka.strimzi.io/v1beta2
+ * kind: KafkaNodePool
+ * metadata:
+ *   name: broker
+ *   labels:
+ *     strimzi.io/cluster: my-cluster
+ * spec:
+ *   replicas: 3
+ *   roles:
+ *     - broker
+ *   storage:
+ *     type: jbod
+ *     volumes:
+ *       - id: 0
+ *         type: persistent-claim
+ *         size: 100Gi
+ *         deleteClaim: false
+ *       - id: 1
+ *         type: persistent-claim
+ *         size: 200Gi
+ *         deleteClaim: false
+ * ---
+ *
+ * Using this information, this class generates Cruise Control BrokerCapacityFileResolver config file like the following:
+ *
+ * {
+ *   "brokerCapacities": [
+ *     {
+ *       "brokerId": "0",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log0": "100000",
+ *           "/var/lib/kafka1/kafka-log0": "200000"
+ *         },
+ *         "CPU": { "num.cores": "2.345" },
+ *         "NW_IN": "10000",
+ *         "NW_OUT": "40000"
+ *       },
+ *       "doc": "Capacity for Broker 0"
+ *     },
+ *     {
+ *       "brokerId": "1",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log1": "100000",
+ *           "/var/lib/kafka1/kafka-log1": "200000"
+ *         },
+ *         "CPU": { "num.cores": "4" },
+ *         "NW_IN": "60000",
+ *         "NW_OUT": "20000"
+ *       },
+ *       "doc": "Capacity for Broker 1"
+ *     },
+ *     {
+ *       "brokerId": "2",
+ *       "capacity": {
+ *         "DISK": {
+ *           "/var/lib/kafka0/kafka-log2": "100000",
+ *           "/var/lib/kafka1/kafka-log2": "200000"
+ *         },
+ *         "CPU": { "num.cores": "4" },
+ *         "NW_IN": "60000",
+ *         "NW_OUT": "20000"
+ *       },
+ *       "doc": "Capacity for Broker 2"
+ *     }
+ *   ]
+ * }
+ */
+public class CapacityConfiguration {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(CapacityConfiguration.class.getName());
+
+    private final TreeMap<Integer, CapacityEntry> capacityEntries;
+
+    /**
+     * Constructor
+     *
+     * @param reconciliation        Reconciliation marker.
+     * @param spec                  Spec of the Kafka custom resource.
+     * @param kafkaBrokerNodes      List of the broker nodes which are part of the Kafka cluster.
+     * @param kafkaStorage          A map with storage configuration used by the Kafka cluster and its node pools.
+     * @param kafkaBrokerResources  A map with resource configuration used by the Kafka cluster and its broker pools.
+     */
+    public CapacityConfiguration(
+            Reconciliation reconciliation,
+            KafkaSpec spec,
```

Comment:
Is it needed to pass the entire Kafka spec and not just the Cruise Control one? AFAICS we need only Cruise Control spec and this is what we get when inside the constructor we call the method for generating the configuration.

Reply:
Fixed.